boot/zephyr/include/mcuboot_config/mcuboot_config.h

```diff
@@ -430,7 +430,9 @@
 #endif
 
 /* Support 32-byte aligned flash sizes */
-#if DT_HAS_CHOSEN(zephyr_flash)
+#if defined(CONFIG_MCUBOOT_BOOT_MAX_ALIGN) && CONFIG_MCUBOOT_BOOT_MAX_ALIGN > 0
+#define MCUBOOT_BOOT_MAX_ALIGN CONFIG_MCUBOOT_BOOT_MAX_ALIGN
+#elif DT_HAS_CHOSEN(zephyr_flash)
```

Comment:
We have these dt_ macros to use with Kconfig, can we maybe move the entire logic to Kconfig?

Reply:
Am I missing something here?

Reply:
As a side-note, we're backwards compatible like this indeed, but not forwards compatible (which was te case before my change).
So if this change is pulled in without https://github.com/zephyrproject-rtos/zephyr/pull/96001 the behavior changes.